File: tests/src/main/cxx/TestScreen.cpp
#include "Test.hpp"

#include <cstdlib>

TestScreen::TestScreen(char const * path)
    : _path(path)
{
}

void TestScreen::takeScreenshot()
{
    std::stringstream cmd;

    cmd << SCREENSHOT_CMD " > "
        << _path.string()
        << "/cxx/"
        << _id_screenshot;

    (void)system(cmd.str().c_str());

    _id_screenshot = _id_screenshot + 1;
}

File: src/test/scala/TestScreen.scala
import scalanative.libc.stdlib.system
import scalanative.unsafe.*

// NOTE: Currently, cannot use JUnit's rules to retrieve the test name -> Use of `testName` variable
@SuppressWarnings(Array("org.wartremover.warts.Var"))
class TestScreen(var folderPath: String = ".", var testName: String = ""):
    private var idScreenshot = 0

    final def takeScreenshot(): Unit =
        val path = s"${folderPath}/scala/screenshot_${idScreenshot}"
        val cmd = s"xwd -root -silent | dd bs=3184 iflag=fullblock skip=1 status=none of=${path}"

        Zone { implicit z => system(toCString(cmd)) }
        idScreenshot = idScreenshot + 1
